src/runtime_src/core/tools/common/ReportAie.cpp

```diff
@@ -0,0 +1,163 @@
+  aie_array.push_back(std::make_pair("", populate_aie(_pDevice, "aie_metadata", "Aie_Metadata")));
+
+  // There can only be 1 root node
+  _pt.add_child("aie_metadata", aie_array);
+}
+
+void 
+ReportAie::writeReport( const xrt_core::device * _pDevice,
+                                  const std::vector<std::string> & /*_elementsFilter*/, 
+                                  std::iostream & _output) const
+{
+  boost::property_tree::ptree _pt;
+  boost::property_tree::ptree empty_ptree;
+  getPropertyTreeInternal(_pDevice, _pt);
+
+  _output << "Aie\n";
+  boost::property_tree::ptree& aies = _pt.get_child("aie_metadata", empty_ptree);
+  for(auto& kv : aies) {
+    boost::property_tree::ptree& aie_data = kv.second;
+    _output << boost::format("  %-10s\n") % aie_data.get<std::string>("description");
+        boost::property_tree::ptree& pt = aie_data.get_child("data", empty_ptree);
+        try {
+            for (auto& gr: pt.get_child("aie_metadata.graphs")) {
+                std::string name,col,row,memcol,memrow,id,memaddr;
+                boost::property_tree::ptree& graph = gr.second;
+                _output << boost::format("  GRAPH[%2d] Name:%2s\n") % graph.get<std::string>("id") % graph.get<std::string>("name");
+                _output << boost::format("    SNo.  %-20s%-30s%-30s\n") % "Core [C:R]" % "Iteration_Memory [C:R]" % "Iteration_Memory_Addresses";
+                int count = 0;
+                auto row_it = gr.second.get_child("core_rows").begin();
+                auto memcol_it = gr.second.get_child("iteration_memory_columns").begin();
+                auto memrow_it = gr.second.get_child("iteration_memory_rows").begin();
+                auto memaddr_it = gr.second.get_child("iteration_memory_addresses").begin();
+                for (auto& node : gr.second.get_child("core_columns")) {
+                    col = node.second.data();
+                    row = row_it->second.data();
+                    memcol = memcol_it->second.data();
+                    memrow = memrow_it->second.data();
+                    memaddr = memaddr_it->second.data();
+                    _output << boost::format("    [%2d]   %-20s%-30s%-30d\n") % count % (col+":"+row) % (memcol+":"+memrow) % memaddr;
+                    row_it++;memcol_it++;memrow_it++;memaddr_it++;count++;
+                }
+                _output << std::endl;
+
+                _output << boost::format("    %s\n") % "Pl Kernel Instances in Graph:";
+                std::string plname;
+                for (auto& node : gr.second.get_child("pl_kernel_instance_names")) {
+                    plname = node.second.data();
+                    _output << boost::format("      %s\n") % plname;
+                }
+                _output << std::endl;
+            }
+
+            int count = 0;
+            for (auto& rtp_node : pt.get_child("aie_metadata.RTPs")) {
+                _output << boost::format("  %-3s:[%2d]\n") % "RTP" % count;

```

Comment:
Are any of the "get"s optional? If they are, then please use:
`get<std::string>("port_name", "N/A")`
So that the try block doesn't fail

Reply:
few strings are empty because the are empty string in original aie_metadata xml in xclbin

Reply:
I see. If they are empty, we should add "N/A" as the default value